state/indexer/query_range.go

```diff
@@ -87,14 +87,13 @@ func (qr QueryRange) UpperBoundValue() interface{} {
 func LookForRangesWithHeight(conditions []query.Condition) (queryRange QueryRanges, indexes []int, heightRange QueryRange) {
 	queryRange = make(QueryRanges)
 	for i, c := range conditions {
-		heightKey := false
+		heightKey := c.CompositeKey == types.BlockHeightKey || c.CompositeKey == types.TxHeightKey
```

Comment:
```suggestion
		
```

Reply:
The reason why I removed it here before the `RangeOperation` check is that it might still be a `height` key but not a range operation. That is then yet another special case we need to handle. Otherwise the fix is legit ,thanks a lot for catching and fixing this!